File: acousticslib_matlab/hann_fades.m
function x_fade = hann_fades(x, num_samples_per_fade)
window = hann(num_samples_per_fade*2).';
x_fade = x;
x_fade(1:num_samples_per_fade) = x_fade(1:num_samples_per_fade).*window(1:num_samples_per_fade);
x_fade(length(x_fade) - (num_samples_per_fade - 1) : end) = x_fade(length(x_fade) - (num_samples_per_fade - 1) : end).*window(num_samples_per_fade + 1 : end);
end

File: utils/acousticslib/hann_fades.py
from numpy import multiply, hanning
def hann_fades(x, num_samples_per_fade):
    window = hanning(num_samples_per_fade*2).T
    x_fade = x.copy()
    x_fade[:num_samples_per_fade] = multiply(x_fade[:num_samples_per_fade], window[:num_samples_per_fade])
    x_fade[len(x_fade) - (num_samples_per_fade) :] = multiply(x_fade[len(x_fade) - (num_samples_per_fade):], window[num_samples_per_fade:])
    return x_fade